src/aux-server/aux-web/aux-auth/site/AuthApp/AuthApp.ts

```diff
@@ -127,47 +147,103 @@ export default class AuthApp extends Vue {
     startCreateStudio() {
         this.showCreateStudio = true;
         this.studioName = '';
+        this.showEnterStudioNameError = false;
+        this.showCreateStudioError = false;
     }
 
     startCreateRecord(studioId?: string) {
         this.showCreateRecord = true;
         this.recordName = '';
         this.createRecordStudioId = studioId ?? '';
+        this.showEnterRecordNameError = false;
+        this.showInvalidRecordNameError = false;
     }
 
     async createStudio() {
-        this.showCreateStudio = false;
-        const comId = authManager.getComIdFromUrl();
-        const result = await authManager.client.createStudio({
-            displayName: this.studioName,
-            ownerStudioComId: comId,
-        });
-        await this.loadStudios();
+        try {
+            this.showEnterStudioNameError = false;
+            this.showCreateStudioError = false;
+            this.errorMessage = '';
+
+            if (!this.studioName) {
+                this.showEnterStudioNameError = true;
+                return;
+            }
+
+            const comId = authManager.getComIdFromUrl();
+            const response = await authManager.client.createStudio({
+                displayName: this.studioName,
+                ownerStudioComId: comId,
+            });
+
+            if (response.success == false) {
+                this.errorMessage = response.errorMessage;
+                if (response.errorCode === 'invalid_key') {
+                    this.showCreateStudioError = true;
+                }
+                return;
+            }
```

Comment:
@TroyceGowdy Please prefer using `getFormErrors()` with an errors array (of `FormError`) instead of checking error codes manually. You can add the `record_already_exists` error code to the [`fieldMap` in FormError.ts](https://github.com/casual-simulation/casualos/blob/a97f792355c3ffb569c759309a7a2810512aa346/src/aux-records/forms/FormError.ts#L44-L58) and then use the [FieldErrors](https://github.com/casual-simulation/casualos/blob/e7ff3679133d7b92ef5bbcf6af0a6cc79fd75a8f/src/aux-server/aux-web/shared/vue-components/FieldErrors/FieldErrors.ts) component to display the errors for the field.

The examples I posted yesterday should have each of those steps.